### ISSUE TITLE
Being able to select a previous kernel at boot time

### ISSUE THREAD

**Issue by mabicca** [Microsoft/azserialconsole#3]
It seems that the serial console loads after the kernel , it would be nice if we could implement a way to have control over grub to select a previous kernel or even change kernel command lines at boot time.

Not sure if that is indeed being implemented that way, where it loads up only if the kernel loads successfully , there are some changes that should be able to give us that scenario, depending how we are implementing it.

A few possible ways to implemente it:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/sec-grub_2_over_a_serial_console

https://help.ubuntu.com/community/SerialConsoleHowto

This could be really helpful in situations of failed to boot after a kernel upgrade as well as kernel crashes during boot.

**Comment by amitchat** [comment]
@mabicca - This is a good point, we looked into a bit and there is a definitely manual steps that can be taken to enable this in your VM and the serial console will allow you to access GRUB.

The challenge is enabling this by default would potentially affect VM start times which makes this a difficult choice at this point.

Thanks for your feedback on this issue.

**Comment by mabicca** [comment]
Hi Amit,

Yes, I understand that we would need to include a timeout I guess correct?

Thank you for the feedback!
-Marco

**Comment by BKButter** [comment]
Hello all,

I used the following work around to access the grub menu.

If you have the option to restart the VM you are fine.

First option: 
1) Option serial console
2) Restart VM
3) Go back to the Serial console. when connection breaks hit “enter button” and then keep on tapping the space bar.

Second option:
I had only the option, to startup the VM ( no restart option) and this did not give me the time needed to halt the startup and enter the Grub menu, the time I got a serial console screen the server is already past the grub menu.

Workaround.
1) Startup the VM server
2) Open a serial console menu
3) Use Powershell to initiate restart from the VM server.
Restart-AzureRmVM -Name “name” -ResourceGroupName “resourcegroup”

4) Go back to the Serial console. when the connection breaks hit “enter button” and then keep on tapping the space bar.
